### PR TITLE
[Core][CLI] `ray status` and `ray memory` no longer starts a new job

### LINKED ISSUE #13250: `ray status` and `ray memory` shouldn't need to start a new job
### Describe your feature request

Because we have a limited number of job ids, running `watch -n 1 ray memory` (arguably the intended use case of the command), we will run out of job ids in ~18 hours. 

We can get all the information we need using `GlobalStateAccessor` without calling `ray.init`. 

For feature requests or questions, post on our Discussion page instead: https://discuss.ray.io/

cc @ericl

### PR DESCRIPTION
**Pull request by micahtyong**
## Why are these changes needed?

This PR resolves [this issue](https://github.com/ray-project/ray/issues/13250) by accessing the `GlobalStateAccessor` directly in `ray memory` and `ray status` without calling `ray.init()` directly and starting a new job. Original problem statement: 

> Because we have a limited number of job ids, running watch -n 1 ray memory (arguably the intended use case of the command), we will run out of job ids in ~18 hours.

## Related issue number

Closes #13250

## Checks

- [x] I've run `scripts/format.sh` to lint the changes in this PR.
- [x] I've included any doc changes needed for https://docs.ray.io/en/master/.
- [x] I've made sure the tests are passing. Note that there might be a few flaky tests, see the recent failures at https://flakey-tests.ray.io/
- Testing Strategy
   - [x] Unit tests
   - [ ] Release tests
   - [ ] This PR is not tested :(

cc: @rkooo567

### CODE REVIEW

**Comment by micahtyong** [comment]
@wuisawesome, this resolves the issue you brought up earlier [here](https://github.com/ray-project/ray/issues/13250). Hope this helps!

File: python/ray/autoscaler/_private/commands.py [hunk 1]
```diff
@@ -90,10 +90,11 @@ def try_reload_log_state(provider_config: Dict[str, Any],
         return reload_log_state(log_state)
 
 
-def debug_status() -> str:
+def debug_status(status=None, error=None) -> str:
     """Return a debug string for the autoscaler."""
-    status = _internal_kv_get(DEBUG_AUTOSCALING_STATUS)
-    error = _internal_kv_get(DEBUG_AUTOSCALING_ERROR)
+    if status is None and error is None:
```
Comment (wuisawesome) [inline python/ray/autoscaler/_private/commands.py:95]: Does this case ever happen? If not, can we just get rid of this code and make the args required?

**Comment by wuisawesome** [comment]
Thanks so much for this! Just left one comment.

File: python/ray/autoscaler/_private/commands.py (hunk 1 above)
Comment (micahtyong) [reply python/ray/autoscaler/_private/commands.py]: You're right, `commands.py#debug_status` is only ever called by the `ray status` CLI. Resolved now!

**Review by wuisawesome — APPROVED** [review]
LGTM, the last lint failure looked unrelated (just reran it to turn it green, but you may need to re-pull master). 

Thanks for this contribution!

**Pull request merged**